sdk/cosmosdb/cosmos/src/retry/timeoutFailoverRetryPolicy.ts

```diff
@@ -0,0 +1,100 @@
+import { RetryPolicy } from "./RetryPolicy";
+import { StatusCodes } from "../common/statusCodes";
+import { GlobalEndpointManager } from "../globalEndpointManager";
+import { HTTPMethod, isReadRequest } from "../common";
+import { ErrorResponse } from "../request";
+import { Constants, ResourceType } from "../common/constants";
+import { RetryContext } from "./RetryContext";
+import { CosmosHeaders } from "../queryExecutionContext/CosmosHeaders";
+import { OperationType } from "@azure/cosmos";
+import { TimeoutErrorCode } from "../request/TimeoutError";
+
+export class TimeoutFailoverRetryPolicy implements RetryPolicy {
```

Comment:
Can you pls add detailed comments about how this policy should behave.

Reply:
done

Reply:
Why are we not using values defined in [https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/cosmosdb/cosmos/src/retry/retryOptions.ts](url) to initialize some of these values here?

Also can we move the default values for these to [https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/cosmosdb/cosmos/src/common/constants.ts](url) so that all constants are at one place and use those constants here to initialize values.